src/vconsole/vconsole-setup.c

```diff
@@ -54,6 +54,55 @@ static bool is_vconsole(int fd) {
         return ioctl(fd, TIOCLINUX, data) >= 0;
 }
 
+static bool is_allocated(int fd, int idx) {
+        struct vt_stat vcs = {};
+        char vcname[strlen("/dev/vcs") + DECIMAL_STR_MAX(int)];
+
+        if (fd >= 0) {
+                if (ioctl(fd, VT_GETSTATE, &vcs) < 0) {
+                        (void) log_warning_errno(errno, "VT_GETSTATE failed: %m");
+                        return false;
+                }
+                idx = vcs.v_active;
+        }
+
+        xsprintf(vcname, "/dev/vcs%i", idx);
+        if (access(vcname, F_OK) < 0)
+                return false;
+        return true;
+}
+
+static bool is_settable(int *fd, int idx) {
+        char vcname[strlen("/dev/tty") + DECIMAL_STR_MAX(int)];
+        int fdt, r, curr_mode;
+
+        if (*fd < 0) {
+                /* try to open terminal */
+                xsprintf(vcname, "/dev/tty%i", idx);
+                fdt = open_terminal(vcname, O_RDWR|O_CLOEXEC);
+                if (fdt < 0)
+                        return false;
+        } else
+                fdt = *fd;
+
+        r = ioctl(fdt, KDGKBMODE, &curr_mode);
+        /*
+         * Make sure we only adjust consoles in K_XLATE or K_UNICODE mode.
+         * Oterwise we would (likely) interfere with X11's processing of the
+         * key events.
+         *
+         * http://lists.freedesktop.org/archives/systemd-devel/2013-February/008573.html
+         */
+        if (r != 0 || (curr_mode != K_XLATE && curr_mode != K_UNICODE)) {
+                if (*fd < 0)
+                        (void) close(fdt);
+                return false;
+        }
+
+        *fd = fdt;
+        return true;
+}
```

Comment:
Hmmm, this piece of code is already outdated, the function is much shorter now and doesn't do any descriptor opening/closing anymore.